examples/05_scaling_non_linear_models.py

```diff
@@ -0,0 +1,537 @@
+r"""
+Fitting scalable, non-linear models on data with dirty categories
+=================================================================
+
+A very classic dilemna when training a machine learning model consists in
+choosing between using a linear model or a non-linear one.
+
+Linear models are very well studied and understood. They are generally fast
+and easy to optimize, and generate interpretable results. However, for some
+problems with a complex relationship between the input and the output, linear
+models reach their expressivity limit: whatever the number of samples the
+training set may have, past some point, it's precision won't get any better.
+
+Non-linear models, however, tend to *scale* better with sample size: they are
+able to digest the information in the additional samples to get a better
+estimate of the link between the input and the output. This idea is illustrated
+in this plot:
+
+.. image:: ../_static/model_capacity.png
+   :align: center
```

Comment:
I would remove this image, and rather point to a relevant page.

In general, I would try to summarize a bit the whole discussion here, to be more concise. However, if you want, I can do it myself after the merge.

Reply:
I removed the image. Feel free to change the text after the merge, it will probably be faster that me iterating several times upon your feedback.